src/Neo.CLI/CLI/MainService.Tools.cs

```diff
@@ -445,6 +571,214 @@ private string ClearHexString(string hexString)
             }
         }
 
+        /// <summary>
+        /// Converts a Base64 string to Hex string.
+        /// input: SGVsbG8gV29ybGQh
+        /// output: 48656c6c6f20576f726c6421
+        /// </summary>
+        /// <param name="base64">
+        /// String that represents the Base64 value
+        /// </param>
+        /// <returns>
+        /// Returns null when the string does not represent a Base64 value or when
+        /// it is not possible to parse the Base64 value to Hex string; otherwise
+        /// returns the string that represents the converted Hex string
+        /// </returns>
+        public static string? Base64ToHexString(string base64)
+        {
+            try
+            {
+                var bytes = Convert.FromBase64String(base64.Trim());
+                return bytes.ToHexString();
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Converts a public key to neo3 address.
+        /// input:  03dab84c1243ec01ab2500e1a8c7a1546a26d734628180b0cf64e72bf776536997
+        /// output: Nd9NceysETPT9PZdWRTeQXJix68WM2x6Wv
+        /// </summary>
+        /// <param name="pubKey">public key</param>
+        /// <returns>Returns null when the string does not represent a public key; otherwise
+        /// returns the string that represents the converted neo3 address.
+        /// </returns>
+        public string? PublicKeyToAddress(string pubKey)
+        {
+            pubKey = pubKey.ToLower();
+            if (!new Regex("^(0[23][0-9a-f]{64})+$").IsMatch(pubKey)) return null;
+            return Contract.CreateSignatureContract(ECPoint.Parse(pubKey, ECCurve.Secp256r1)).ScriptHash.ToAddress(NeoSystem.Settings.AddressVersion);
+        }
+
+        /// <summary>
+        /// Converts a Private key to public key
+        /// </summary>
+        /// <param name="wif">Private key in WIF format</param>
+        /// <returns>Returns null when the string does not represent a private key; otherwise
+        /// returns the string that represents the converted public key.
+        /// </returns>
+        public static string? PrivateKeyToPublicKey(string wif)
```

Comment:
wif to pubkey？

Reply:
> wif to pubkey？

Yeah. What's the problem?

Reply:
nothing，just thinking maybe you can make it both support wif and privatkey

Reply:
It worries me that people use a private key so openly, why not create a wallet?